numba/core/ir_utils.py

```diff
@@ -662,6 +672,7 @@ def has_no_side_effect(rhs, lives, call_table):
             call_list == ['dtype', numpy] or
             call_list == [array_analysis.wrap_index] or
             call_list == [prange] or
+            call_list == ['prange', numba] or
```

Comment:
Is there any way of creating a test case for this, purely so that it doesn't accidentally get broken in the future?

Reply:
Done.